### DISCUSSION TITLE
`engine.raw_connection` now raises unwrapped DBAPI exceptions

### DISCUSSION THREAD

**Discussion by TomaszParadysz-SlashDB** [sqlalchemy/sqlalchemy#12104] [Usage Questions]
Hi,
I noticed that since SQLAlchemy 2.0, the `engine.raw_connection` method now raises unwrapped DBAPI exceptions, while in 1.4 it raised SQLAlchemy exceptions. I didn't find any mention of such change in the changelog.

Example code to show the issue:
```py
from sqlalchemy import create_engine

connstr = 'postgresql://user:password@127.0.0.1:5432/DB'

engine = create_engine(connstr)

try:
    engine.raw_connection()
except Exception as exc:
    print(type(exc))
    print(exc)
```

1.4.47 output:
```
<class 'sqlalchemy.exc.OperationalError'>
(psycopg2.OperationalError) connection to server at "127.0.0.1", port 5432 failed: FATAL:  password authentication failed for user "user"

(Background on this error at: https://sqlalche.me/e/14/e3q8)
```

2.0.36 output:
```
<class 'psycopg2.OperationalError'>
connection to server at "127.0.0.1", port 5432 failed: FATAL:  password authentication failed for user "user"
```
---

My question is: Is this an intentional change and I should work around that, or is this a bug and I should expect it to be fixed?

Thanks in advance

**Comment by zzzeek** [comment]
this got changed when engine was refactored for 2.0 in d050193da but was never noted in any changelog or test suite.

I think this is because the usual way to get at a DBAPI connection is to use the [pattern in the docs](https://docs.sqlalchemy.org/en/20/core/connections.html#dbapi-connections) where you `.connect()` first then work with `.connection`.   all the methods you call from the DBAPI connection are going to raise driver level errors anyway.

it's not clear what `raw_connection()` should do but we can accept a PR for documentation

**Reply by CaselIT** [reply]
I think the 2.0 behaviour is better, since the return type is not an sqlalchemy connection, so it follows that it does not raise sqlalchemy exceptions

**Reply by TomaszParadysz-SlashDB** [reply]
This method is, to me, on the border between sqlalchemy and DBAPI, so I could see how it could make sense either way (wrapped vs unwrapped).
But, I think that doing `engine.connect().connection` should be fine in my case, so I'll just go with that.
Thanks for the response

**Reply by CaselIT** [reply]
I prefer unwrapped since if you do `engine.raw_connection().execute('something')` and the execute errors you get an unwrapped exception.

But I agree that it would be fine either way, the issue is that there was no changelog
